test/modules/ecs6-class/line.test.js

```diff
@@ -0,0 +1,160 @@
+const Line = require("../../../modules/ecs6-class/line")
+const Point = require("../../../modules/ecs6-class/point")
+
+describe("LINE_FUNCTION", function () {
+    describe("BUILD_LINE", function () {
+        it("should show line after build the line", function () {
+            let p1 = new Point({ x: 1, y: 2 })
+            let p2 = new Point({ x: 3, y: 4 })
+            let point1 = new Point({ x: 5, y: 5 })
+            let point2 = new Point({ x: 6, y: 6 })
+            let l1 = new Line({ point1: p1, point2: p2, n: 2, slope: -3 })
+            let l2 = new Line({ point1: point1, point2: point2, })
+            let l3 = new Line({ point1: undefined, point2: undefined })
+            expect(l3.n).toBe(undefined)
+            expect(l1.point1.x).toEqual(1)
+            expect(l1.point1.y).toEqual(2)
+            expect(l1.point2.x).toEqual(3)
+            expect(l1.point2.y).toEqual(4)
+            expect(l1.n).toEqual(2)
+            expect(l1.slope).toEqual(-3)
+            expect(l2.point1.x).toEqual(5)
+            expect(l2.point1.y).toEqual(5)
+            expect(l2.point2.x).toEqual(6)
+            expect(l2.point2.y).toEqual(6)
+            expect(l2.n).toBe(undefined)
+            expect(l2.slope).toBe(undefined)
+
+
+        })
+    })
+    describe("CALCULATE_SLOPE", function () {
+        it("should calculate the slope between 2 points", function () {
+            let p1 = new Point({ x: 1, y: 2 })
+            let p2 = new Point({ x: 3, y: 4 })
+            let point1 = new Point({ x: 5, y: 5 })
+            let point2 = new Point({ x: 6, y: 6 })
+            let l1 = new Line({ point1: p1, point2: p2, n: 2, slope: -3 })
+            let l2 = new Line({ point1: point1, point2: point2, })
+            expect(l1.slope).toEqual(-3)
+
+            l1.calculateSlope()
+            l2.calculateSlope()
+            expect(l1.slope).toEqual(1)
+            expect(l2.slope).toEqual(1)
```

Comment:
when you test two cases, separate it into two tests

Reply:
I don't understand why to divide into 2 cases

Reply:
It's a way of writing code